Neos.Neos/Resources/Private/Fusion/Prototypes/Menu.fusion

```diff
@@ -1,48 +1,18 @@
 # Neos.Neos:Menu provides basic menu rendering
 #
-prototype(Neos.Neos:Menu) < prototype(Neos.Fusion:Template) {
-  templatePath = 'resource://Neos.Neos/Private/Templates/FusionObjects/Menu.html'
-
-  entryLevel = ${this.startingPoint ? 0 : 1}
-  maximumLevels = 2
-  startingPoint = null
-  lastLevel = null
-  filter = 'Neos.Neos:Document'
-  renderHiddenInIndex = false
-  itemCollection = null
-
-  node = ${node}
-
+prototype(Neos.Neos:Menu) < prototype(Neos.Fusion:Component) {
   attributes = Neos.Fusion:DataStructure
 
-  active.attributes = Neos.Fusion:DataStructure {
-    class = 'active'
-  }
-  current.attributes = Neos.Fusion:DataStructure {
-    class = 'current'
-  }
-  normal.attributes = Neos.Fusion:DataStructure {
-    class = 'normal'
-  }
-
-  @context {
-    entryLevel = ${this.entryLevel}
-    maximumLevels = ${this.maximumLevels}
-    startingPoint = ${this.startingPoint}
-    lastLevel = ${this.lastLevel}
-    filter = ${this.filter}
-    renderHiddenInIndex = ${this.renderHiddenInIndex}
-    itemCollection = ${this.itemCollection}
+  @private {
+    items = Neos.Neos:MenuItems {
+      @ignoreProperties = ${['attributes']}
+      @apply.props = ${props}
+    }
```

Comment:
Hmm this `@apply.props = ${props}` is quite a clever pattern. But im torn apart whether to use it here. 

We lately put some work into it to define for every prototype its defaults https://github.com/neos/neos-development-collection/pull/3915.
And since the amount of properties for the menu api are final i like to propose to do here the same (again).

Reply:
Yeah. Would do that aswell but during the untangling of the menu prototype inheritance. So in a later pr.

Reply:
Okay sound like a good plan ;)

Then let’s do it later 👍